### ISSUE TITLE
Scrolling by mouse?

### ISSUE THREAD

**Issue by Wandalen** [rhysd/tui-textarea#2]
Hello! Thanks for the great Open Source!
Is scrolling by mouse implemented?
Can I do it without editing your code?

**Comment by rhysd** [comment]
This crate has not yet supported scrolling by mouse. Scrolling only happens when a cursor moves outside the textarea for now.

A workaround I can come up with is moving a cursor on scroll. But the behavior would be different from general mouse scrolling and look strange.

```rust
if let Event::Mouse(e) = crossterm::event() {
    if let MouseEventKind::Down = e {
        textarea.move_cursor(CursorMove::Down);
    }
}
```

Thank you for feature request. I'll add this to my TODO list.

**Comment by rhysd** [comment]
I tried to implement the workaround in `editor` example at [this branch](https://github.com/rhysd/tui-textarea/tree/mouse-workaround).


```sh
git clone https://github.com/rhysd/tui-textarea.git
git checkout mouse-workaround
cargo run --example editor --features=search README.md
```

**Comment by Wandalen** [comment]
oh, amazing! :)

**Comment by rhysd** [comment]
Mouse scroll was implemented at 4d6ad3d. The workaround will be no longer necessary at next release.

**Comment by Wandalen** [comment]
oh, amazing! :)
